src/components/HistoryTable/HistoryTableTestSearch/HistoryTableTestSearch.tsx

```diff
@@ -1,75 +1,83 @@
   onSubmit = () => {},
 }) => {
   const [input, setInput] = useState("");
+  const isValid = validateRegexp(input);
+  const handleSubmit = useUpsertQueryParams();
 
-  const updateQueryParams = useUpdateURLQueryParams();
-  const { search } = useLocation();
-  const queryParams = parseQueryString(search);
-
-  // Currently, users are only able to filter by failed tests
-  const onClick = () => {
-    const selectedParams = queryParams.failed as string[];
-    const updatedParams = upsertQueryParam(selectedParams, input);
-    onSubmit(updatedParams);
-    updateQueryParams({ failed: updatedParams });
-    setInput("");
+  const handleOnChange = (value: string) => {
+    setInput(value);
+  };
+  const handleOnSubmit = () => {
+    if (isValid) {
+      onSubmit();
+      handleSubmit({ category: TestStatus.Failed, value: input });
+      setInput("");
+    }
   };
 
   return (
     <ContentWrapper>
-      <TextInputWrapper>
-        <TextInput
-          type="search"
-          label="Filter by Failed Tests"
-          aria-label="history-table-test-search-input"
-          value={input}
-          placeholder="Search test name regex"
-          onChange={(e) => setInput(e.target.value)}
-          onKeyPress={(e) => e.key === "Enter" && onClick()}
-        />
-        <StyledIcon glyph="Plus" onClick={() => onClick()} />
-      </TextInputWrapper>
+      <TextInput
+        type="search"
+        label="Filter by Failed Tests"
+        aria-label="history-table-test-search-input"
+        value={input}
+        placeholder="Search test name regex"
+        onChange={(e) => handleOnChange(e.target.value)}
+        onKeyPress={(e: React.KeyboardEvent<HTMLInputElement>) =>
+          e.key === "Enter" && handleOnSubmit()
+        }
+        icon={
+          isValid ? (
+            <IconButton
+              onClick={handleOnSubmit}
+              aria-label="Select plus button"
+            >
+              <Icon glyph="Plus" data-cy="tuple-select-button" />
+            </IconButton>
+          ) : (
+            <InactiveIconWrapper>
+              <IconTooltip
+                glyph="Warning"
+                data-cy="tuple-select-warning"
+                fill={yellow.base}
+              >
+                Invalid Regular Expression
+              </IconTooltip>
+            </InactiveIconWrapper>
+          )
+        }
+      />
     </ContentWrapper>
   );
 };
 
 const ContentWrapper = styled.div`
-  display: flex;
-  flex-direction: column;
   width: 40%;
   min-width: 200px; //temporary
-  padding-right: 30px;
-`;
-
-const TextInputWrapper = styled.div`
+  margin-right: 30px;
   position: relative;
-  display: flex;
-  flex-direction: column;
-  justify-content: center;
 `;
 
-const StyledIcon = styled(Icon)`
-  position: absolute;
-  height: 100%;
-  top: 10px; //temporary
-  align-self: flex-end;
-  margin-right: 10px;
-  &:hover {
-    cursor: pointer;
-  }
+const InactiveIconWrapper = styled.div`
+  margin: 6px;
```

Comment:
same comment

Reply:
Nice catch remnants of my n'thousandth iteration at making this reusable.

Reply:
reminder to remove this style as it's no longer necessary

Reply:
oof.